rero_ils/modules/patrons/api.py

```diff
@@ -440,18 +440,20 @@ def get_circulation_messages(self, public=False):
         # if patron is blocked - error type message
         #   if patron is blocked, no need to return any other circulation
         #   messages !
-        if self.is_blocked:
+        if not public and self.is_blocked:
             return [{
                 'type': 'error',
                 'content': self.get_blocked_message(public)
             }]
 
         messages = []
         # if patron expiration_date has reached - error type message
-        if self.is_expired:
+        if not public and self.is_expired:
             messages.append({
                 'type': 'error',
-                'content': _('Patron rights expired.')
+                'content': _('Your account has expired. Please contact'
+                             ' your library.') if public else
+                _('Patron rights expired.')
```

Comment:
you have problem in this code : First `if` statement includes `not public` and your ternary instruction use `if public`.